### PR TITLE
[docs] Documentation restructure

### PR DESCRIPTION
**Pull request by bmorelli25**
For https://github.com/elastic/apm-agent-python/issues/616. Demo available **[here](http://apm-agent-python_617.docs-preview.app.elstc.co/guide/en/apm/agent/python/master/index.html)**.

I'd like to backport this to `4.x`. For `5.x`, I'm guessing the best way is to just wait for the next release?

- [x] Top level heading and doc sync for log correlation: https://github.com/elastic/apm/issues/149
- [x] Use shared version attributes: https://github.com/elastic/apm/issues/150
- [x] Add a section on how the Agent does instrumentation: https://github.com/elastic/apm/issues/146
- [x] Rename the APM UI to "APM app": https://github.com/elastic/apm-dev/issues/567
- [x] Use `apm-app-ref`: https://github.com/elastic/docs/pull/1276
- [x] Updates to the new documentation structure outlined [here](https://docs.google.com/spreadsheets/d/1P53LbNY7jFSLLJhOKw-nmDenQbrIFlY4EEFJGqT1kmY/edit#gid=0)
- [x] Sync Central config docs: https://github.com/elastic/apm-agent-dotnet/pull/497#discussion_r326303325

Screen Shot 2019-10-22 at 4 34 56 PM

### CODE REVIEW

File: docs/getting-started.asciidoc [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
+like HTTP requests and database queries. To do this, it {CONTENT_HERE}
+This means that for the supported technologies, there are no code changes required.
 
-The Elastic APM Python agent sends performance metrics and error logs to the APM Server.
-It has built-in support for Django and Flask performance metrics and error logging, as well as generic support of other WSGI frameworks for error logging.
+The Agent automatically {CONTENT_HERE}
+
+These events, called Transactions and Spans, are sent to the APM Server.
+The APM Server converts them to a format suitable for Elasticsearch, and sends them to an Elasticsearch cluster.
+You can then use the APM app in Kibana to gain insight into latency issues and error culprits within your application.
```
Comment (bmorelli25) [inline docs/getting-started.asciidoc]: One of the things this PR aims to do is add in a section on how the agent does instrumentation. I've added some boilerplate text here to hopefully assist with this. There are two {CONTENT_HERE} sections above that I'll need one of you to fill out. Feel free to write something different or amend this as necessary.

I've also updated https://github.com/elastic/apm/issues/146 with links to examples in other Agent's docs.

**Comment by bmorelli25** [comment]
@beniwohli, think you'll have time for a review on this one and https://github.com/elastic/apm-agent-python/pull/619 at some point next week?

**Comment by beniwohli** [comment]
@bmorelli25 I'll take a look today, and try to carve out some time tomorrow to write the "how we instrument" section.

File: docs/flask.asciidoc [hunk 2]
```diff
@@ -174,7 +174,7 @@ app = Flask(__name__)
 
 apm = ElasticAPM(app)
 
-if __name__ == '__main__':
+if __name__ === '__main__':
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/flask.asciidoc]: oh-bother

**Comment by bmorelli25** [comment]
Moving the conversation to this PR:

@beniwohli, I moved the majority of your content to the _Advanced topics_ section. I kept just few high-level paragraphs in the getting started section. Here's what I kept (feel free to edit)

```asciidoc
[[how-it-works]]
=== How does the Agent work?

The Python Agent instruments your application to collect APM events in a few different ways:

To collect data about incoming requests and background tasks, the Agent integrates with <<supported-technologies,supported technologies>> to make use of hooks and signals provided by the framework.
These framework integrations require limited code changes in your application.

To collect data from database drivers, HTTP libraries etc.,
we instrument certain functions and methods in these libraries.
Instrumentations are set up automatically and do not require any code changes.

In addition to APM and error data,
the Python agent also collects system and application metrics in regular intervals.
This collection happens in a background thread that is started by the agent.

More detailed information on how the Agent works can be found in <<link-here>>.

```

cc @basepi

**Comment by basepi** [comment]
@bmorelli25 I think your changes are 👍 👍 -- it was definitely too technical for a Getting Started page, but valuable for a click-through, so this is perfect.

> Instrumentations are set up automatically and do not require any code changes.

This is the only piece I'm unsure on -- technically you do need code changes, you have to import our library or add us to the INSTALLED_APPS -- maybe something like this?

> Instrumentations are set up automatically and do not require any code changes, beyond the initial top-level setup.

^ I don't love that but I'm trying to stay succinct. Or maybe I'm being too picky on the details?

**Comment by bmorelli25** [comment]
That's a good point @basepi. I had originally copied the "Instrumentations are set up automatically and do not require any code changes." sentece directly from Beni's PR. I like the change here though. I'll make it tomorrow unless @beniwohli has any objections.

**Comment by bmorelli25** [comment]
Follow up question -- anything I can do to get the `continuous-integration/appveyor/pr — AppVeyor build failed` check to pass?

**Pull request merged**